lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -17,38 +17,61 @@ class FlexChargeGateway < Gateway
         sync: 'outcome',
         refund: 'orders/%s/refund',
         store: 'tokenize',
-        inquire: 'orders/%s'
+        inquire: 'orders/%s',
+        capture: 'capture'
       }
 
-      SUCCESS_MESSAGES = %w(APPROVED CHALLENGE SUBMITTED SUCCESS PROCESSING).freeze
+      SUCCESS_MESSAGES = %w(APPROVED CHALLENGE SUBMITTED SUCCESS PROCESSING CAPTUREREQUIRED).freeze
 
       def initialize(options = {})
         requires!(options, :app_key, :app_secret, :site_id, :mid)
         super
       end
 
       def purchase(money, credit_card, options = {})
+        options[:transactionType] ||= 'Purchase'
+
         post = {}
-        address = options[:billing_address] || options[:address]
         add_merchant_data(post, options)
         add_base_data(post, options)
         add_invoice(post, money, credit_card, options)
         add_mit_data(post, options)
-        add_payment_method(post, credit_card, address, options)
-        add_address(post, credit_card, address)
+        add_payment_method(post, credit_card, address(options), options)
+        add_address(post, credit_card, address(options))
         add_customer_data(post, options)
         add_three_ds(post, options)
 
         commit(:purchase, post)
       end
 
+      def authorize(money, credit_card, options = {})
+        options[:transactionType] = 'Authorization'
+        purchase(money, credit_card, options)
+      end
+
+      def capture(money, authorization, options = {})
+        order_id, currency = authorization.split('#')
+        post = {
+          idempotencyKey: options[:idempotency_key] || SecureRandom.uuid,
+          orderId: order_id,
+          amount: money,
+          currency: currency
+        }
+
+        commit(:capture, post, authorization)
+      end
+
       def refund(money, authorization, options = {})
-        commit(:refund, { amountToRefund: (money.to_f / 100).round(2) }, authorization)
+        order_id, _currency = authorization.split('#')
+        commit(:refund, { amountToRefund: (money.to_f / 100).round(2) }, order_id)
```

Comment:
Is there a reason to format the `amountToRefund` here as opposed to setting the `self.money_format` and calling the `localized_amount` method to format it?

Reply:
Nop, I wasn't aware and  just saw a rounding operation and did it, going to add the suggested change thanks 👍